lib/matplotlib/container.py

```diff
@@ -26,6 +26,7 @@ def remove(self):
                 self, scalarp=lambda x: isinstance(x, Artist)):
             if c is not None:
                 c.remove()
+        self.set_label("")
```

Comment:
I guess this is OK - it's a bit strange that legend is treating a Container as an Artist, but "remove" doesn't remove the Artist from consideration by legend.   Setting the label to be empty accomplishes legend ignoring the Container, but at this point the Container should not be showing up in the list of Artists.

Reply:
well, you would think, right? But I guess for some reason, it still shows up in the legend unless you specifically set the label to empty. Weird, right?

Reply:
Yes - but I think this would be better if it chased down why that doesn't happen.  Just blanking the label papers over the problem, but isn't a very robust fix.  For instance you could imagine at some point we decide to include legend entries for artists with empty label strings.

Reply:
@jklymak, the bug has nothing to do with auto-updating; as the test cases show, the auto-updating likely has to do with the flattening of containers. but it likely has something to do with flattening the nested containers, in which would likely be of greater risk for something else happening since "cbook.flatten" is referenced so much. I looked into other ways of removing it from the legend, but it seems as though there are very strict Object-Oriented protocols that I did not want to violate in my pull request this seemed most in line with the current syntax of the rest of the function. The label also is not present if you query how many labels are present in the value.

Reply:
My point is that the container is apparently being left in the list of artists on the axis, and it should not be.

Reply:
I'd look at how we get the list of handles in `_get_legend_handles` and decide why the the Collection has not been removed from `ax._children`.  Note I'm not suggesting changing anything in legend, probably just in `Collection.remove`

Reply:
Looking at it more, I'd just put a bunch of print statements in `container.remvoe` and try and figure out why `self._remove_method(self)` is either not being called or not working